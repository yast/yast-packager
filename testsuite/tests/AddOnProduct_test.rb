# encoding: utf-8

# Testsuite for Addon.ycp module
#
module Yast
  class AddOnProductTestClient < Client
    def main
      Yast.include self, "testsuite.rb"

      # huh, we need to mock too much paths because of some module constructor... :-(
      @READ = {
        "target"    => {
          "tmpdir" => "/tmp",
          "size"   => 1,
          "stat"   => { "isreg" => true }
        },
        "xml"       => {},
        "sysconfig" => {
          "language" => {
            "RC_LANG"             => "en_US.UTF-8",
            "ROOT_USES_LANG"      => "ctype",
            "RC_LANG"             => "en_US.UTF-8",
            "INSTALLED_LANGUAGES" => ""
          },
          "console"  => { "CONSOLE_ENCODING" => "UTF-8" }
        }
      }

      @EXEC = {
        "target" => {
          "bash_output" => { "exit" => 0, "stdout" => "charmap=\"UTF-8\"\n" }
        }
      }

      TESTSUITE_INIT([@READ, {}, @EXEC], nil)

      Yast.import "AddOnProduct"

      @url = "http://example.com/repos/SLES11SP2"

      # invalid input
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(nil, nil, nil) }, [], nil)

      # keep the url untouched
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(@url, "", "") }, [], nil)
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(@url, nil, nil) }, [], nil)

      # set name as alias, alias is empty
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(@url, "", "SLES-11-SP2") }, [], nil)

      # set name as alias, alias is empty
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(@url, "SLES", "SLES-11-SP2") }, [], nil)



      # an alias is already set in URL

      @url = "http://example.com/repos/SLES11SP2?alias=mySLES"

      # keep the url untouched
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(@url, "", "") }, [], nil)

      # keep the original alias
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(@url, "", "SLES-11-SP2") }, [], nil)

      # set alias
      TEST(lambda { AddOnProduct.SetRepoUrlAlias(@url, "SLES", "SLES-11-SP2") }, [], nil)

      nil
    end
  end
end

Yast::AddOnProductTestClient.new.main
