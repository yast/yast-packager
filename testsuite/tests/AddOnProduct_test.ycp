/**
 * Testsuite for Addon.ycp module
 *
 */

{

include "testsuite.ycp";

// huh, we need to mock too much paths because of some module constructor... :-(
map READ = $[
  "target" : $[
    "tmpdir" : "/tmp",
    "size" : 1,
    "stat" : $["isreg": true],
    "string" : "SUSE Linux Enterprise Server 11 (x86_64)\nVERSION = 11\nPATCHLEVEL = 1\n"
  ],
  "xml" : $[],
  "sysconfig" : $[
    "language" : $[
	"RC_LANG" : "en_US.UTF-8",
	"ROOT_USES_LANG" : "ctype",
	"RC_LANG" : "en_US.UTF-8",
	"INSTALLED_LANGUAGES" : ""
    ],
    "console" : $[
	"CONSOLE_ENCODING" : "UTF-8"
    ]
  ]
];

map EXEC = $[
    "target" : $[
	"bash_output" : $[ "exit" : 0, "stdout" : "charmap=\"UTF-8\"\n"]
    ]
];

TESTSUITE_INIT ([READ, $[], EXEC], nil);

import "AddOnProduct";

string url = "http://example.com/repos/SLES11SP2";

// invalid input
TEST(``(AddOnProduct::SetRepoUrlAlias(nil, nil, nil)), [], nil);

// keep the url untouched
TEST(``(AddOnProduct::SetRepoUrlAlias(url, "", "")), [], nil);
TEST(``(AddOnProduct::SetRepoUrlAlias(url, nil, nil)), [], nil);

// set name as alias, alias is empty
TEST(``(AddOnProduct::SetRepoUrlAlias(url, "", "SLES-11-SP2")), [], nil);

// set the alias
TEST(``(AddOnProduct::SetRepoUrlAlias(url, "SLES", "SLES-11-SP2")), [], nil);



// an alias is already set in URL

url = "http://example.com/repos/SLES11SP2?alias=mySLES";

// keep the url untouched
TEST(``(AddOnProduct::SetRepoUrlAlias(url, "", "")), [], nil);

// keep the original alias
TEST(``(AddOnProduct::SetRepoUrlAlias(url, "", "SLES-11-SP2")), [], nil);

// set alias, override the value from URL
TEST(``(AddOnProduct::SetRepoUrlAlias(url, "SLES", "SLES-11-SP2")), [], nil);

}

