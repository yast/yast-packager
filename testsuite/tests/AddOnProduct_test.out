Read	.target.tmpdir "/tmp"
Read	.target.stat "/y2update/control.xml" $["isreg":true]
Read	.target.size "/y2update/control.xml" 1
Read	.xml "/y2update/control.xml" $[]
Read	.sysconfig.console.CONSOLE_ENCODING "UTF-8"
Execute	.target.bash_output "locale -k charmap" $["exit":0, "stdout":"charmap=\"UTF-8\"\n"]
Read	.sysconfig.language.RC_LANG "en_US.UTF-8"
Read	.target.size "/etc/sysconfig/language" 1
Read	.sysconfig.language.ROOT_USES_LANG "ctype"
Read	.sysconfig.language.RC_LANG "en_US.UTF-8"
Read	.sysconfig.language.INSTALLED_LANGUAGES ""
Log	Invalid 'url' parameter: nil
Return	nil
Return	http://example.com/repos/SLES11SP2
Return	http://example.com/repos/SLES11SP2
Return	http://example.com/repos/SLES11SP2
Return	http://example.com/repos/SLES11SP2?alias=SLES
Return	http://example.com/repos/SLES11SP2?alias=mySLES
Return	http://example.com/repos/SLES11SP2?alias=SLES-11-SP2
Return	http://example.com/repos/SLES11SP2?alias=SLES
